src/shared/nuclearnet/nuclearnet_proxy_parser.ts

```diff
@@ -35,8 +36,17 @@ export class Encoder {
           case 'unlisten':
             return callback([JSON.stringify(packet)])
 
+          case 'packet': {
+            const { id, data: [key, { target, type, payload, reliable }] } = packet
+            return callback([
+              JSON.stringify({ id, nsp, key, header: { target, type, reliable } }),
```

Comment:
You're sending `type` as a string here, and adding a hashType below that isn't a part of a NUClearNetSend packet.
The decoding logic below will take that and add a .hash element to it afterwards that'll be ignored, so the hashType(type) here is doing nothing.
However, NUClearNetSend does accept a buffer or a string here, so maybe instead don't send the type in json and in the parser below add it as the type.

Reply:
Updated the decoder to not add the unused hash.